### DISCUSSION TITLE
DynamicModel key sort order

### DISCUSSION THREAD

**Discussion by lestan** [pocketbase/pocketbase#6200] [Q&A]
I'm trying to store data in a JSON field in a table.  

I'm using DynamicField to populate the data, however I'm noticing that the keys are sorted alphabetically when inserted into the table.  How do I preserve the ordering?  Is there a way to pass a Map object?

My scenario is to store AI chat messages which have a keys 'human' and 'ai' to represent user messages and AI assistant messages.  I need to store it with human, then AI.  When using DynamicModel, it stores it as AI, then human.  

Appreciate any guidance.  Thanks in advance.

**Comment by ganigeorgiev** [comment]
It is not possible to preserve the keys because the Go std json encoding normalizes the order. If the order is important use an array or add sorted prefix to your keys.

**Reply by lestan** [reply]
Thanks Gani
